Feature Request: Sort balances

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What problem does your feature solve?

Balances cam come in different order between request, the following responses contain the same data but the order is different in both:

```
"balances": [
    {
      "balance": "2.0100000",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 14599295,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "SLT",
      "asset_issuer": "GCKA6K5PCQ6PNF5RQBF7PQDJWRHO6UOGFMRLK3DYHDOI244V47XKQ4GP"
    },
    {
      "balance": "1.0000000",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 11479636,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "HUG",
      "asset_issuer": "GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP"
    },
    {
      "balance": "0.0099999",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 10135487,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "EUR",
      "asset_issuer": "GDI73WJ4SX7LOG3XZDJC3KCK6ED6E5NBYK2JUBQSPBCNNWEG3ZN7T75U"
    },
    {
      "balance": "0.0024939",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000002",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 19967410,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "DEMO",
      "asset_issuer": "GBAMBOOZDWZPVV52RCLJQYMQNXOBLOXWNQAY2IF2FREV2WL46DBCH3BE"
    },
    {
      "balance": "0.0000000",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 20511785,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "BTC",
      "asset_issuer": "GAUTUYY2THLF7SGITDFMXJVYH3LHDSMGEAKSBU267M2K7A3W543CKUEF"
    },
    {
      "balance": "90.2366572",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000001",
      "asset_type": "native"
    }
  ]
```

vs

```
"balances": [
    {
      "balance": "2.0100000",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 14599295,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "SLT",
      "asset_issuer": "GCKA6K5PCQ6PNF5RQBF7PQDJWRHO6UOGFMRLK3DYHDOI244V47XKQ4GP"
    },
    {
      "balance": "1.0000000",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 11479636,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "HUG",
      "asset_issuer": "GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP"
    },
    {
      "balance": "0.0000000",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 20511785,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "BTC",
      "asset_issuer": "GAUTUYY2THLF7SGITDFMXJVYH3LHDSMGEAKSBU267M2K7A3W543CKUEF"
    },
    {
      "balance": "0.0099999",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 10135487,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "EUR",
      "asset_issuer": "GDI73WJ4SX7LOG3XZDJC3KCK6ED6E5NBYK2JUBQSPBCNNWEG3ZN7T75U"
    },
    {
      "balance": "0.0024939",
      "limit": "922337203685.4775807",
      "buying_liabilities": "0.0000002",
      "selling_liabilities": "0.0000000",
      "last_modified_ledger": 19967410,
      "is_authorized": true,
      "asset_type": "credit_alphanum4",
      "asset_code": "DEMO",
      "asset_issuer": "GBAMBOOZDWZPVV52RCLJQYMQNXOBLOXWNQAY2IF2FREV2WL46DBCH3BE"
    },
    {
      "balance": "90.2366572",
      "buying_liabilities": "0.0000000",
      "selling_liabilities": "0.0000001",
      "asset_type": "native"
    }
  ]
```

### What would you like to see?

I'd like for balances to keep the same order between requests. It could be trustlines first (as it is today), sorted by `asset_code` and `native` at the end.


### What alternatives are there?

Not sorting

I'm working on a completely unrelated code and just got a failure in `TestAccountInfo`:
```
--- FAIL: TestAccountInfo (0.72s)
    db.go:122: Test Database: test_4e220c223c3f
    db.go:122: Test Database: test_785e362bfd9c
    account_test.go:233: 
        	Error Trace:	account_test.go:233
        	Error:      	Not equal: 
        	            	expected: "USD"
        	            	actual  : "EUR"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-USD
        	            	+EUR
        	Test:       	TestAccountInfo
    account_test.go:234: 
        	Error Trace:	account_test.go:234
        	Error:      	Not equal: 
        	            	expected: "EUR"
        	            	actual  : "USD"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-EUR
        	            	+USD
        	Test:       	TestAccountInfo
```
Looks like we don't use a new `GetSortedTrustLinesByAccountIDs` in `actions.AccountInfo` (still used by legacy actions code).

--- a/services/horizon/internal/actions/account.go
+++ b/services/horizon/internal/actions/account.go
@@ -39,7 +39,7 @@ func AccountInfo(ctx context.Context, hq *history.Q, addr string) (*protocol.Acc
 		return nil, errors.Wrap(err, "getting history signers")
 	}
 
-	trustlines, err = hq.GetTrustLinesByAccountID(addr)
+	trustlines, err = hq.GetSortedTrustLinesByAccountID(addr)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting history trustlines")
 	}

--- a/services/horizon/internal/db2/history/trust_lines.go
+++ b/services/horizon/internal/db2/history/trust_lines.go
@@ -32,11 +32,8 @@ func (q *Q) CountTrustLines() (int, error) {
 	return count, nil
 }
 
-func (q *Q) GetTrustLinesByAccountID(id string) ([]TrustLine, error) {
-	var trustLines []TrustLine
-	sql := selectTrustLines.Where(sq.Eq{"account_id": id})
-	err := q.Select(&trustLines, sql)
-	return trustLines, err
+func (q *Q) GetSortedTrustLinesByAccountID(id string) ([]TrustLine, error) {
+	return q.GetSortedTrustLinesByAccountIDs([]string{id})
 }
 
 // GetTrustLinesByKeys loads a row from the `trust_lines` table, selected by multiple keys.

--- a/services/horizon/internal/db2/history/trust_lines_test.go
+++ b/services/horizon/internal/db2/history/trust_lines_test.go
@@ -352,7 +352,7 @@ func TestGetTrustLinesByAccountID(t *testing.T) {
 	_, err := q.InsertTrustLine(eurTrustLine, 1234)
 	tt.Assert.NoError(err)
 
-	record, err := q.GetTrustLinesByAccountID(eurTrustLine.AccountId.Address())
+	record, err := q.GetSortedTrustLinesByAccountID(eurTrustLine.AccountId.Address())
 	tt.Assert.NoError(err)
 
 	asset := xdr.MustNewCreditAsset(record[0].AssetCode, record[0].AssetIssuer)

